Add clarification for otel-js apps written in CJS and ESM (#5072)
open-telemetry/opentelemetry.io: content/en/docs/languages/js/getting-started/nodejs.md

- [TypeScript](https://www.typescriptlang.org/download), if you will be using
  TypeScript.

{{% alert title="Note" color="info" %}}

> Much of OpenTelemetry JS documentation is written assuming the compiled application is run as CommonJS.

Reviewer: ```suggestion
The OpenTelemetry documentation assumes that the compiled application is run as CommonJS.
```

Would it make sense to add a link to CommonJS? Are all Node developers familiar with it?

{{% /tab %}} {{< /tabpane >}}

> ### Instrumentation for ECMAScript Modules

Reviewer: ```suggestion
### Instrumentation for ECMAScript modules
```

> If your application is written in JavaScript as ESM, or compiled to ESM from TypeScript, then a loader hook is required to properly patch instrumentation.
> The custom hook for ESM instrumentation is `--experimental-loader=@opentelemetry/instrumentation/hook.mjs`.
> This flag must be passed to the `node` binary, which is often done as a startup command and/or in the `NODE_OPTIONS` environment variable.

Reviewer: ```suggestion
If your application is written in JavaScript as ESM, or compiled to ESM
from TypeScript, a loader hook is required to patch instrumentation.
The custom hook for ESM instrumentation is
`--experimental-loader=@opentelemetry/instrumentation/hook.mjs`.
This flag must be passed to the `node` binary, which is often done
as a startup command or through the `NODE_OPTIONS` environment variable.
```

```console
$ node --experimental-loader=@opentelemetry/instrumentation/hook.mjs --require ./instrumentation.js app.js
Listening for requests on http://localhost:8080
```

> 

Reviewer 2: Having the h3 section header for this unintentionally puts the subsequent content ("Open http://localhost:8080/rolldice in your web browser ...") under this section.

I don't have a great alternative suggestion. Perhaps the section could be replaced with a much shorter:

> (Note: If your application is written in JavaScript as ECMAScript Modules (ESM), or compiled to ESM from TypeScript, then a loader hook is required to properly support instrumentation. Use `node --experimental-loader=@opentelemetry/instrumentation/hook.mjs --require ./instrumentation.js app.js`. See [the coming reference link in opentelemetry-js/docs/] for details on ESM support.)
